lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueService.kt

```diff
@@ -19,19 +24,27 @@ class HmppsQueueService {
   fun findByDlqName(dlqName: String) = hmppsQueues.associateBy { it.dlqName }.getOrDefault(dlqName, null)
 
   fun retryDlqMessages(request: RetryDlqRequest): RetryDlqResult =
-    with(request.hmppsQueue) {
-      val messageCount = sqsAwsDlqClient.countMessagesOnQueue(dlqUrl)
-      val messages = mutableListOf<Message>()
-      repeat(messageCount) {
-        sqsAwsDlqClient.receiveMessage(ReceiveMessageRequest(dlqUrl).withMaxNumberOfMessages(1)).messages.firstOrNull()
-          ?.let { msg ->
-            sqsAwsClient.sendMessage(queueUrl, msg.body)
-            sqsAwsDlqClient.deleteMessage(DeleteMessageRequest(dlqUrl, msg.receiptHandle))
-            messages += msg
-          }
-      }
-      return RetryDlqResult(messageCount, messages.toList())
+    request.hmppsQueue.retryDlqMessages()
+
+  fun retryAllDlqs() =
+    hmppsQueues
+      .map { hmppsQueue -> RetryDlqRequest(hmppsQueue) }
+      .map { retryDlqRequest -> retryDlqMessages(retryDlqRequest) }
```

Comment:
should this be  a `run` or whatever, first look at code things like you are trying to return the `retryDlqMessages`

Reply:
It returns a list of `RetryDlqResult`.  I'm not totally sure what to do with it yet... but I'm thinking of just sending it back to the caller.